### ISSUE TITLE
Problema plugin com OSC e cupom de desconto

### ISSUE THREAD

**Issue by nelsonbf** [gumnet/magento-1-ame-digital-integration#4]
Olá,

Encontramos um problema no uso do plugin com cupom de desconto. Checkout utilizado, OSC da Inovarti.
Ao fazer uma compra sem cupom de desconto, o valor apresentado na tela de finalização, que gera o qr code da AME, o valor total aparece corretamente.

Agora, se eu tenho um cupom aplicado, ele pega o total da compra, diminui mais uma vez o cupom e apresenta o valor reduzido ao cliente. No entanto, só apresenta, pq a cobrança é feita corretamente. 

Exemplo no print abaixo:

O valor que tinha que ter sido apresentado era o total de 344,70. Ele diminui 38,30 (que já havia sido descontado) e apresentou 306,40

![image](https://user-images.githubusercontent.com/114436796/192356842-615a82d6-d097-4366-a545-02415fc07754.png)

<img width="329" alt="image" src="https://user-images.githubusercontent.com/114436796/192357054-469f254b-076a-413b-9f0e-6af7cf4fd50d.png">

**Comment by LuizSantos22** [comment]
@nelsonbf O senhor está usando o plugin com Magento 1 ou OpenMage? Pergunto porque uso OpenMage com o OSC da Moip ou o padrão e nenhum dos dois gera o QR Code. 

Vou até testar novamente pra ver se mudou algo...

Att
Luiz Santos

**Comment by gustavoulyssea** [comment]
@nelsonbf obrigado por abrir este caso, confirmo que conversamos e vou avaliar o caso e te dar um retorno. Por favor me dê um a dois dias e te mando a confirmação.

**Comment by gustavoulyssea** [comment]
@nelsonbf peço desculpas, ainda não conseguimos focar neste caso. Como o Magento 1 acabou sendo abandonado em troca pelo Magento 2 acabamos mantendo a prioridade do calendário para o Magento 2. Se puder dar uma sugestão avalie esta migração pois aos poucos os métodos de pagamento param de dar suporte para o software obsoleto além das questões de falta de atualização pelo abandono da Adobe.

**Comment by LuizSantos22** [comment]
Gente, M1 foi abandonado, mas Openmage passou a ser a continuidade e é
adotado por muitos lojistas. Tem mais de 100k lojistas no mundo.

Em sex, 28 de out de 2022 12:32, gustavoulyssea ***@***.***>
escreveu:

**Comment by gustavoulyssea** [comment]
Respeito as opiniões pessoais sobre software independente, mas o software citado não é o Magento, é apenas uma tentativa de seguir com o funcionamento de um software abandonado há alguns anos.

Entendo que é difícil começar a programar numa plataforma nova visto que o Magento 2 foi todo refeito, mas assim funciona a tecnologia, diariamente precisamos estar nos atualizando.

**Comment by LuizSantos1** [comment]
@nelsonbf 
Isso pode ser um problema com a OSC da Inovarti, não do módulo em si, pois tive no passado um problema semelhante com desconto que não era aplicado corretamente, com este módulo de OSC... Busque pelo módulo aqui no github e dê uma olhada nas issues que provavelmente você vai encontrar algo parecido. Tente testar no checkout padrão ou o OSC da IWD ou da Moip pra ver se o problema segue. 

Quanto à discussão sobre OM (M1) e M2
Bom, a OM (OpenMage) não tem o intuito de substituir a M2, e não é uma "tentativa" de continuidade do M1, ela é uma fork do M1, há muitos anos, inclusive antes mesmo da Adobe adquirir a Magento. Outro exemplo de fork, é a InovarTI, que é uma fork da OM em si.

Seria como comparar um canivete comum (OM) com um canivete suiço (M2), porém, o canivete comum é mais fácil de mantê-lo afiado e não requer muitos recursos para que ele funcione. Portanto, pode servir para o propósito de médios e pequenos lojistas. O canivete suiço, pode ser excelente pra lojistas de médio pra grande porte, grande porte e gigantes, mas um desastre em termos de custos de manutenção para os lojistas de pequeno porte. Ou seja, são softwares diferentes, que atendem propósitos e clientes diferentes.  

Softwares equiparáveis à OM são Shopify, Opencart, NuvenShop, etc... 

Não há como dizer que um software onde a última release de segurança e melhorias de performance, inclusive com adoção de PHP 8.1 que ocorreu agora e 04 de Setembo, está "abandonada". Cada um atende a plataforma que achar mais conveniente, porém, menos ativismo, galera! Um abraço!

**Comment by gumnet** [comment]
Aqui não é um local para discussão de gostos e opiniões, é um local para notificar o desenvolvedor sobre problemas técnicos.

Os comentários foram removidos. Favor respeitar o espaço.

Pull requests com melhorias são bem-vindos. Serão avaliados, e podem ser aprovados dependendo do conteúdo e da melhoria.

Muito obrigado.

**Comment by LuizSantos1** [comment]
> > 
> 
> Plugin com Magento 1.

Dê uma checada no checkout padrão do Magento, o problema pode estar no OSC da Inovarti, que é antigo. Já tive esse tipo de problema no passado.

**Comment by LuizSantos1** [comment]
> Aqui não é um local para discussão de gostos e opiniões, é um local para notificar o desenvolvedor sobre problemas técnicos.
> 
> Os comentários foram removidos. Favor respeitar o espaço.
> 
> Pull requests com melhorias são bem-vindos. Serão avaliados, e podem ser aprovados dependendo do conteúdo e da melhoria.
> 
> Muito obrigado.

@gumnet Concordo em parte, já que não existe qualquer regra no Github onde as pessoas
são proibidas de emitir opiniões e apontar possíveis soluções nos comentários.
E foi exatamente o que eu fiz. Divergi e corrigir o comentário do desenvolvedor que alegou
que OM seria uma plataforma obsoleta, o que não é verdade, não é falta de respeito.

Na mesma divergência, apontei uma possível solução à issue levantada, sugerindo que o 
problema possa estar no OSC da InovarTi em si, este sim, já obsoleto. 

Divergencia de opiniões nunca será falta de respeito, silenciamento dos divergentes, talvez.
Github é um espaço open source onde as contribuições, mesmo que divergentes
deveriam ser abraçadas, não suprimidas.

Um abraço!

**Comment by gustavoulyssea** [comment]
Olá Nelson,

Tudo bom? Encontrei um pequeno bug e corrigi, por favor dá uma olhada neste arquivo atualizado:

https://github.com/gumnet/magento-1-ame-digital-integration/blob/master/app/design/frontend/base/default/template/ame/qrcode.phtml

Obrigado pela paciência.

Atenciosamente,
Gustavo Ulyssea

**Comment by nelsonbf** [comment]
Obrigado Gustavo
